internal/pkg/cli/job_init_test.go

```diff
@@ -0,0 +1,101 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package cli
+
+import (
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/spf13/afero"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestJobInitOpts_Validate(t *testing.T) {
+	testCases := map[string]struct {
+		inJobName        string
+		inDockerfilePath string
+		inTimeout        string
+		inRetries        int
+		inSchedule       string
+
+		mockFileSystem func(mockFS afero.Fs)
+		wantedErr      error
+	}{
+		"invalid job name": {
+			inJobName: "1234",
+			wantedErr: fmt.Errorf("job name 1234 is invalid: %s", errValueBadFormat),
+		},
+		"invalid dockerfile directory path": {
+			inDockerfilePath: "./hello/Dockerfile",
+			wantedErr:        errors.New("open hello/Dockerfile: file does not exist"),
+		},
+		"invalid schedule; cron and rate": {
+			inSchedule: "every 56 minutes",
+			wantedErr:  errors.New("schedule value every 56 minutes must be either a Go duration string or a valid cron expression"),
+		},
+		"invalid schedule; rate too frequent": {
+			inSchedule: "7s",
+			wantedErr:  errors.New("schedule rate 7s must be greater than a minute"),
+		},
+		"invalid schedule; rate in subseconds": {
+			inSchedule: "75.9s",
+			wantedErr:  errors.New("schedule rate 75.9s cannot be in units smaller than a second"),
+		},
+		"valid schedule; cron": {
+			inSchedule: "* * * * *",
+			wantedErr:  nil,
+		},
+		"valid schedule; rate": {
+			inSchedule: "1h23m45s",
+			wantedErr:  nil,
+		},
+		"invalid timeout duration; incorrect format": {
+			inTimeout: "30 minutes",
+			wantedErr: errors.New("time: unknown unit  minutes in duration 30 minutes"),
```

Comment:
Where does this error arise? I wonder if it might be better for us to wrap it with some context: `%s is not a valid duration string: %w`

Reply:
Good idea. It comes from the time package, from ParseDuration.